### ISSUE TITLE
mybe a bug in finetune.py

### ISSUE THREAD

**Issue by xiexie123** [kratzert/finetune_alexnet_with_tensorflow#32]
hi
    I found my Validation Accuracy was a constant value during 10 epochs or enven 40 epochs. I am very confused.
   I found in Validate:
        sess.run(validation_init_op)
        .....
             img_batch, label_batch = sess.run(next_batch)

but the next_batch only find in 
    iterator = Iterator.from_structure(tr_data.data.output_types,
                                       tr_data.data.output_shapes)
    next_batch = iterator.get_next()

so, is there any possibility a bug here, and should add like this:
        iterator_val = Iterator.from_structure(val_data.data.output_types,
                                       val_data.data.output_shapes)
       next_batch_val = iterator_val.get_next()

**Comment by xiexie123** [comment]
I add 
iterator_val = Iterator.from_structure(val_data.data.output_types,
val_data.data.output_shapes)

next_batch_val = iterator_val.get_next()
validation_init_op = iterator_val.make_initializer(val_data.data)

run finetune.py is ok ,but still  my Validation Accuracy is a constant value .... 

anyone knows why?....

**Comment by kratzert** [comment]
this is no bug at all. Maybe you should have a look at the tensorflow documentation to understand what happes there. The only thing that is done in this step is to tell the iterator what kind of data and in what shape it will see. If you look at the very next lines you could see that we make a tensorflow op to initialize the iterator on the validation data (validation_init_op = iterator.make_initializer(val_data.data)) and this is called at the beginning of the validation period. 

So error must be in your data or what ever you changed in the code. Maybe you are not training any layer at all, then the validation accuracy would stay the same all the time.

**Comment by xiexie123** [comment]
my mistake. I will check my code. Thank you
